libraries/tools/kotlin-main-kts/src/org/jetbrains/kotlin/mainKts/scriptDef.kt

```diff
@@ -90,6 +94,72 @@ class MainKtsHostConfiguration : ScriptingHostConfiguration(
     }
 )
 
+fun configureScriptFileLocationPathVariablesForEvaluation(context: ScriptEvaluationConfigurationRefinementContext): ResultWithDiagnostics<ScriptEvaluationConfiguration> {
+    val scriptId = context.compiledScript.sourceLocationId ?: return context.evaluationConfiguration.asSuccess()
+    val compilationConfiguration = context.evaluationConfiguration[ScriptEvaluationConfiguration.compilationConfiguration]
+        ?: return context.evaluationConfiguration.asSuccess()
+    val scriptFileLocationMap = compilationConfiguration[ScriptCompilationConfiguration.scriptFileLocation]
```

Comment:
It seems that after you solved the problem with link to the compilation configuration in the previous commit, this code could be simplified, I guess you no longer need a map: a simple property in the compilation configuration will be enough. Could you, please check, if it could work?

Reply:
I tried this at the initial stages and it did not work for me for some reason but now I tried again and all seems fine. Thanks for this remark, now indeed the code looks much nicer :)